bitcoin/src/pow.rs

```diff
@@ -229,8 +230,14 @@ impl Target {
     /// [max]: Target::max
     /// [target]: crate::blockdata::block::Header::target
     #[cfg_attr(all(test, mutate), mutate)]
-    pub fn difficulty(&self) -> u128 {
-        let d = Target::MAX.0 / self.0;
+    pub fn difficulty(&self, network: Network) -> u128 {
+        let max = match network {
+            Network::Bitcoin => Target::MAX_ATTAINABLE_MAINNET,
+            Network::Testnet => Target::MAX_ATTAINABLE_TESTNET,
+            Network::Signet => Target::MAX_ATTAINABLE_SIGNET,
+            Network::Regtest => Target::MAX_ATTAINABLE_REGTEST,
+        };
```

Comment:
Feels like this should be in consensus params.

Reply:
These are in `Params::pow_limit`, I thought it better to take a `Network` parameter not a `Params` parameter though because its more specific.

Reply:
We could have this?
```rust
    pub fn difficulty(&self, params: &Params) -> u128 {
        let max = params.pow_limit;
        let d = max.0 / self.0;
        d.saturating_to_u128()
    }
```

Reply:
I do think `Network` makes sense but I was thinking `Params::new(network).pow_limit`.

Reply:
Once we have `Network: AsRef<Params>`/`Network: AsParams` we should just take generic.